image-encapsulate.cpp

```diff
@@ -1,19 +1,24 @@
 #include <iostream>
 #include "brightener.h"
 
-int main() {
-	auto image = std::make_unique<Image>(512, 512);
+int main() 
+{
+	auto image = std::make_shared<Image>(512, 512);
 	std::cout << "Brightening a 512 x 512 image\n";
-	ImageBrightener brightener(std::move(image));
-	if (brightener.ValidateImage()) {
+	std::cout << image.use_count() << std::endl;
```

Comment:
Nice experiment! Even I learnt about `use_count` after seeing your code :)
But better to remove this while delivering... this cout is for our understanding, not related to user-visible functionality

Reply:
removed.